languages/csharp/VaultAPI.cs

```diff
@@ -3,416 +3,600 @@
+                throw new Exception("view not instantiated");
 
-    [DllImport("vaultSDK.dll")]
-    private static extern vdkError vdkPointCloud_GetMetadata(IntPtr pModel, ref string ppJSONMetadata);
-  }
+            vdkError error = vdkRenderView_SetMatrix(pRenderView, matrixType, cameraMatrix);
+            if (error != Vault.vdkError.vE_Success)
+                throw new Exception("vdkRenderView.SetMatrix failed: " + error.ToString());
+        }
 
-  class vdkConvertContext
-  {
-    public void Create(vdkContext context)
-    {
-      vdkError error = vdkConvert_CreateContext(context.pContext, ref pConvertContext);
-      if (error != Vault.vdkError.vE_Success)
-        throw new Exception("vdkConvertContext.Create failed.");
-    }
+        public IntPtr pRenderView = IntPtr.Zero;
+        private vdkContext context;
 
-    public void Destroy()
-    {
-      vdkError error = vdkConvert_DestroyContext(ref pConvertContext);
-      if (error != Vault.vdkError.vE_Success)
-        throw new Exception("vdkConvertContext.Destroy failed.");
-    }
+        private GCHandle colorBufferHandle;
+        private GCHandle depthBufferHandle;
+
+        [DllImport("vaultSDK")]
+        private static extern vdkError vdkRenderView_Create(IntPtr pContext, ref IntPtr ppRenderView, IntPtr pRenderer, UInt32 width, UInt32 height);
+
+        [DllImport("vaultSDK")]
+        private static extern vdkError vdkRenderView_Destroy(ref IntPtr ppRenderView);
+
+        [DllImport("vaultSDK")]
+        private static extern vdkError vdkRenderView_SetTargets(IntPtr pRenderView, IntPtr pColorBuffer, UInt32 colorClearValue, IntPtr pDepthBuffer);
 
+        [DllImport("vaultSDK")]
+        private static extern vdkError vdkRenderView_GetMatrix(IntPtr pRenderView, RenderViewMatrix matrixType, double[] cameraMatrix);
 
-    public void AddFile(string fileName)
+        [DllImport("vaultSDK")]
+        private static extern vdkError vdkRenderView_SetMatrix(IntPtr pRenderView, RenderViewMatrix matrixType, double[] cameraMatrix);
+    }
+
+    public class RenderOptions
     {
-      vdkError error = vdkConvert_AddItem(pConvertContext, fileName);
-      if (error != Vault.vdkError.vE_Success)
-        throw new Exception("vdkConvertContext.AddItem failed.");
+        private unsafe vdkRenderPicking pick;
+        public vdkRenderOptions options;
+        public bool pickSet = false;
+        public bool pickRendered = false;
+
+        public RenderOptions(vdkRenderContextPointMode pointMode, vdkRenderFlags flags)
+        {
+            options.pointMode = pointMode;
+            //this will need to change once support for multiple picks is introduced:
+            options.pPick = Marshal.AllocHGlobal(Marshal.SizeOf(pick));
+            options.flags = flags;
+        }
+
+        public RenderOptions() :this(vdkRenderContextPointMode.vdkRCPM_Rectangles,vdkRenderFlags.vdkRF_None)
+        {
+            
+        }
+
+        public void setPick(uint x, uint y)
```

Comment:
ProperCase for functions

```suggestion
        public void SetPick(uint x, uint y)
```